public/pages/workflows/new_workflow/use_case.tsx

```diff
@@ -90,7 +90,11 @@ export function UseCase(props: UseCaseProps) {
                   .then((result) => {
                     const { workflow } = result;
                     history.replace(
-                      `${APP_PATH.WORKFLOWS}/${workflow.id}?dataSourceId=${dataSourceId}`
+                      `${APP_PATH.WORKFLOWS}/${workflow.id}${
+                        dataSourceId !== undefined
+                          ? `?dataSourceId=${dataSourceId}`
+                          : ''
+                      }`
```

Comment:
can we utilize `constructHrefWithDataSourceId` instead? I see this repeated. Seems very similar. And if not, maybe we can make the helper fn more flexible to handle it, or a separate standalone fn.